### PR TITLE
[Xcode] Set an explicit iOS deployment target for llbuild.

### PR DESCRIPTION
**Pull request by cwakamo**
iOS 9.0 is the oldest version supported by Xcode 13, so this commit sets the deployment target to that.

In support of this, added a missing availability check since iOS now back-deploys instead of deploying to the latest OS only.

This is for <rdar://problem/84568181>.

### CODE REVIEW

**Comment by cwakamo** [comment]
@swift-ci Please smoke test

File: include/llbuild/Basic/Tracing.h [hunk 1]
```diff
@@ -47,17 +47,17 @@ static os_signpost_id_t signpost_id() {
 
 /// Begin an interval if tracing is enabled.
 #define LLBUILD_TRACE_INTERVAL_BEGIN(name, ...) { \
-if (__builtin_available(macOS 10.14, *)) os_signpost_interval_begin(getLog(), signpost_id(), name, ##__VA_ARGS__); \
+if (__builtin_available(macOS 10.14, iOS 12.0, *)) os_signpost_interval_begin(getLog(), signpost_id(), name, ##__VA_ARGS__); \
```
Comment (jakepetroules) [inline include/llbuild/Basic/Tracing.h:50]: Can you add tvOS/watchOS to these three as well? There's also a __builtin_available in adjust-times.cpp that could use adjustment.

**Review by jakepetroules — CHANGES_REQUESTED** [review]

File: include/llbuild/Basic/Tracing.h (hunk 1 above)
Comment (cwakamo) [reply include/llbuild/Basic/Tracing.h:50]: OK, can do -- I'll also set TVOS_DEPLOYMENT_TARGET to 9.0 and WATCHOS_DEPLOYMENT_TARGET to 2.0 to match the iOS deployment target when I fix this.

**Comment by cwakamo** [comment]
@swift-ci Please smoke test

**Review by jakepetroules — APPROVED** [review]

**Pull request merged**